libraries/DateTime/contracts/DateTime.aes

```diff
@@ -114,13 +62,191 @@ contract DateTime =
     let timestamp = timestamp + second  
     timestamp
 
-// Private functions and internal logic
+  public function parse_timestamp(timestamp : int) : date_time = 
+    let dt : date_time = init_date_time(timestamp)
+    
+    let dt : date_time = p_get_year(dt)
+    let dt : date_time = p_get_month(dt)
+    let dt : date_time = p_get_day(dt)
+    let dt : date_time = dt{hour = rest(timestamp / 60 /60, 24)}
+    let dt : date_time = dt{minute = rest(timestamp / 60, 60)}
+    let dt : date_time = dt{second = rest(timestamp, 60)}
+    let wd : int = rest(timestamp / state.day_in_seconds + 4, 7)
+    let dt : date_time = dt{weekday = wd}
+    dt
+
+  public function get_year(timestamp : int) : int = 
+    let year : int = parse_timestamp(timestamp).year
```

Comment:
this can be directly returned, no need to assign to a variable

(true for all cases in this contracts)

Reply:
@HristiyanG how is this resolved?